staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/exec.go

```diff
@@ -323,6 +327,10 @@ func (a *Authenticator) refreshCredsLocked(r *clientauthentication.Response) err
 	}
 
 	if err := cmd.Run(); err != nil {
+		if a.installHint != "" {
+			return fmt.Errorf("exec: %v\n%s", err, cleanANSIEscapeCodes(a.installHint))
+		}
+
```

Comment:
This should only be invoked when the binary does not exist.  If the binary exists and the execution of it fails, the binary should print some helpful text to standard error.

Reply:
I waffled on this. Figured we would have a convo on the PR.

Here was the use case I came up with:
- User is working on unfamiliar box
- Unfamiliar box has an older version of the provider `command` that is on the `$PATH` but execution fails because it is old and crusty
- User wants to see a message of how to install the latest version of the `command`

WDYT?

Reply:
Scratch my ramblings - what you are saying makes sense.

I tried to clean this up in https://github.com/kubernetes/kubernetes/pull/91305/commits/e53aca0c6f8c9a2c82abafd2e7baf6bd0b1c894e.